cunumeric/linalg/cholesky.py

```diff
@@ -12,18 +12,37 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+from __future__ import annotations
 
+from typing import TYPE_CHECKING, cast
 
 from cunumeric.config import CuNumericOpCode
 
 from legate.core import Rect, types as ty
-
-
-def transpose_copy(context, launch_domain, p_input, p_output):
-    task = context.create_task(
-        CuNumericOpCode.TRANSPOSE_COPY_2D,
-        manual=True,
-        launch_domain=launch_domain,
+from legate.core.operation import ManualTask
+
+if TYPE_CHECKING:
+    from legate.core.context import Context
+    from legate.core.shape import Shape
+    from legate.core.store import StorePartition
+
+    from ..deferred import DeferredArray
+    from ..runtime import Runtime
+
+
+def transpose_copy(
+    context: Context,
+    launch_domain: Rect,
+    p_input: StorePartition,
+    p_output: StorePartition,
+) -> None:
+    task = cast(
+        ManualTask,
+        context.create_task(
+            CuNumericOpCode.TRANSPOSE_COPY_2D,
+            manual=True,
+            launch_domain=launch_domain,
+        ),
```

Comment:
I don't think there is a way to narrow the type returned by `create_task`, so this cast is necessary. We could bifurcate `create_task` on the boolean parameter.

Reply:
Let's bifurcate `create_task`. Otherwise, there will be two more ugly castings like this once #391 is merged.

Reply:
OK, I will make create_auto_task and create_manual_task that delegate to create_task but since that will require legate changes I propose to merge this for now, and then make update in the next mypy PR once the legate change is also merged.